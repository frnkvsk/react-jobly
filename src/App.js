import React, { useState } from 'react';
import { BrowserRouter, Route, Switch, useParams } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import NavBar from './components/NavBar';
import Home from './pages/Home';
import Company from './pages/Company';
import Companies from './pages/Companies';
import Jobs from './pages/Jobs';
import Profile from './pages/Profile';
import Login from './pages/Login';
import Page404 from './pages/Page404';
import { SearchContext } from './context/SearchContext';
import { LoginContext } from './context/LoginContext';
import { useLogin } from './hooks/useLogin';

const useStyles = makeStyles((theme) => ({
  root: {
    // height: '100vh',
    backgroundColor: '#f5f5f5',
  },
  
}));
function CompanyHandle() {
  let {handle} = useParams();
  console.log('App CompanyHandle handle=',handle)
  return <Company handle={handle} />
}

function App() {
  const classes = useStyles();
  const [token] = useLogin();
  console.log('App token',token)
  const [search, setSearch] = useState("");
  
  const [loginStatus, setLoginStatus] = useState(token ? true : false);

  return (
      <BrowserRouter>
        <LoginContext.Provider value={{loginStatus, setLoginStatus}}>
          <NavBar loggedin={true} />
          <main className={classes.root}>
            <Switch>
              <Route exact path="/">
                <Home />          
              </Route>
              <Route exact path="/companies/:handle"> 
                <CompanyHandle />
              </Route>
              <Route exact path="/companies">
                <SearchContext.Provider value={{search, setSearch}}>
                  <Companies />
                </SearchContext.Provider>              
              </Route>
              <Route exact path="/jobs">
                <SearchContext.Provider value={{search, setSearch}}>
                  <Jobs />
                </SearchContext.Provider>    
              </Route>
              <Route path="/profile">
                <Profile />
              </Route>
              <Route exact path="/login">
                <Login login={"signup"} />
              </Route>
              <Route>
                <Page404 />
              </Route>
            </Switch>
          </main>
        </LoginContext.Provider>
        
      </BrowserRouter>
  );
}

export default App;
