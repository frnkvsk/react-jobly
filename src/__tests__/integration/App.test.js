import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';
import App from '../../App';

// smoke and appearance tests
test('renders Home ', () => {
  render(<App />);
  
  expect(screen.getAllByText(/jobly/i)[0]).toBeInTheDocument();

  expect(screen.getAllByText(/all the jobs in one, convenient place./i)[0]).toBeInTheDocument();

  expect(screen.getAllByRole('button', {name: /log in \/ sign up/i})[0]).toBeInTheDocument();
});

describe('test App and Login', () => {
  
  beforeAll(async () => {
    render(<App />);
  });
  

  test('can render login page from home page', () => {
    // click 'login / signup' button
    fireEvent.click(screen.getAllByRole('button', {name: /log in \/ sign up/i})[0]);

    // login screen
    expect(screen.getByText(/username/i)).toBeInTheDocument();
    expect(screen.getByText(/password/i)).toBeInTheDocument();
    // username input textbox present
    fireEvent.click(screen.getByRole('textbox', {id: 'username'}), 'testusername1');
    // password input textox present
    fireEvent.click(screen.getByRole('textbox', {id: 'password'}), 'testpassword1');
    // submit button present
    fireEvent.click(screen.getByRole('button', {name: /submit/i}));
    // error message present
    expect(screen.getByText(/Error: Invalid credentials/i)).toBeInTheDocument();
  });
})



// snapshot test
test('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});