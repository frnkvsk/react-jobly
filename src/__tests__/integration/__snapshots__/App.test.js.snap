// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-31"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <a
          class="MuiTypography-root makeStyles-title-33 MuiTypography-h5"
          href="/"
        >
          Jobly
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/login"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            login / signup
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </header>
  </div>
  <main
    class="makeStyles-root-30"
  >
    <div
      class="MuiContainer-root makeStyles-root-35 MuiContainer-maxWidthLg"
    >
      <span
        class="MuiBox-root MuiBox-root-41 makeStyles-err-40"
      >
        Error: Invalid credentials
      </span>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              aria-label="nav tabs example"
              class="MuiTabs-flexContainer"
              role="tablist"
            >
              <a
                aria-disabled="false"
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
                role="tab"
                tabindex="0"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  Login
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
              <a
                aria-disabled="false"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth"
                role="tab"
                tabindex="-1"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  Sign up
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
            <span
              class="PrivateTabIndicator-root-42 PrivateTabIndicator-colorSecondary-44 MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </header>
      <form
        class="makeStyles-form-36"
      >
        <p
          class="MuiFormHelperText-root makeStyles-label-37"
        >
          Username
        </p>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-38"
          variant="outlined"
        >
          <input
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="username"
            name="username"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-46 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-47"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root makeStyles-label-37"
        >
          Password
        </p>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-38"
          variant="outlined"
        >
          <input
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-46 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-47"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <div
          class="makeStyles-button-39"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </main>
</DocumentFragment>
`;
